components/datetime/src/datetime.rs

```diff
@@ -0,0 +1,287 @@
+// This file is part of ICU4X. For terms of use, please see the file
```

Comment:
This file moved out of `components/datetime/src` into its own file module to make room for the `timezone` and `zoned_datetime` modules (and to not have a massive `lib.rs`).